In [4]:
!pip install datasets evaluate accelerate
!pip install sacrebleu jiwer

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 84.1/84.1 kB 919.7 kB/s eta 0:00:000:00:010m
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.0/58.0 kB 965.9 kB/s eta 0:00:00a 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 106.7/106.7 kB 2.1 MB/s eta 0:00:00 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.4/3.4 MB 21.2 MB/s eta 0:00:0000:0100:01


In [5]:
from huggingface_hub import login

login(token="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")

Token has not been saved to git credential helper. Pass `add_to_git_credential=True` if you want to set the git credential as well.
Token is valid (permission: write).
Your token has been saved to /root/.cache/huggingface/token
Login successful


In [6]:
import os
import random
import json
import torch
import torch.nn as nn
from functools import partial
from collections import namedtuple
from dataclasses import dataclass, field, asdict
from huggingface_hub import HfApi

import evaluate
import numpy as np
from datasets import load_dataset
from transformers import Trainer
from transformers import DataCollatorWithPadding
from transformers import AutoTokenizer, TrainingArguments
from datasets import Dataset

import warnings
warnings.filterwarnings("ignore")

2024-06-13 03:01:32.394585: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-06-13 03:01:32.394674: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-06-13 03:01:32.514112: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered


# Download dataset & Preprocess


In [79]:
# Tải bộ dataset
dataset = load_dataset("grammarly/coedit")

In [80]:
dataset

DatasetDict({
    train: Dataset({
        features: ['_id', 'task', 'src', 'tgt'],
        num_rows: 69071
    })
    validation: Dataset({
        features: ['_id', 'task', 'src', 'tgt'],
        num_rows: 1712
    })
})

In [81]:
from datasets import Dataset

In [82]:
import pandas as pd
df1 = dataset['train'].to_pandas()
df2 = dataset['validation'].to_pandas()

df = pd.concat([df1,df2], axis = 0)

In [83]:
# df = dataset['train'].to_pandas()
df = df[df['task'] == 'gec'][['src',  'tgt']].reset_index().drop(columns = 'index')
df['src'] = df['src'].apply(lambda x: x.split(':')[1])
df = df.rename(columns = {'src':'input', 'tgt':'output'})
# dataset['train'] = Dataset.from_pandas(df.copy())

In [84]:
df

input  \
0       For example, countries with a lot of deserts ...   
1       As the number of people grows, the need of ha...   
2       Besides some technologically determinists tha...   
3       Safety is one of the crucial problems that ma...   
4       On one hand more and more virus and hack can ...   
...                                                  ...   
20303   Jane and monica, I receive the letter, first ...   
20304   against THE CINEMA, WE HAVE SEEN A COMEDY ENT...   
20305   The romance is just a small parts where Batma...   
20306   Then, concerning a slight rising in 1992, it ...   
20307   Most of this were down to the speed following...   

                                                  output  
0      For example, countries with a lot of deserts c...  
1      As the number of people grows, the need for a ...  
2      Besides some technological determinists that a...  
3      Safety is one of the crucial problems that man...  
4      On the one hand, more and more viruses and hac...  
...                                                  ...  
20303  Dear Jane and Monica, I have received the lett...  
20304  AT THE CINEMA, WE SAW A COMEDY ENTITLED " AUTU...  
20305  The romance is just a small part where Batman ...  
20306  Then, after a slight rise in 1992, they fell b...  
20307  Most of this is down to the speed at which our...  

[20308 rows x 2 columns]

In [85]:
import pandas as pd
from tqdm import tqdm
import re

# Giả sử df là DataFrame đã được định nghĩa
# df = pd.read_csv("your_file.csv") hoặc bất kỳ phương thức nào bạn dùng để tạo DataFrame

def clean_text(content):
    if not isinstance(content, str):
        return None

    content = content.replace('..', '.') \
                     .replace('"', '') \
                     .replace(': .', '.') \
                     .replace('. .', '.') \
                     .replace("'", '') \
                     .replace('“', '') \
                     .replace('”', '')

#     # Check length of words
#     words = content.split()
#     for word in words:
#         if len(word) > 10:
#             return None

    if content and content[0] == '-':
        content = content.replace('-', '', 1)

    if content and content[0] == ' ':
        content = content[1:]

    content = content.replace('. - ', '. ') \
                     .replace('.. ', '... ') \
                     .replace('?.', '?') \
                     .replace(' ?.', '?') \
                     .replace('( ', '(') \
                     .replace(' )', ')') \
                     .replace('()', '') \
                     .replace('… ', '... ') \
                     .replace('….', '...') \
                     .replace('  ', ' ')

    if content.endswith(' .'):
        content = content[:-2] + '.'

    # Thay thế các ký tự không mong muốn bằng ký tự trống hoặc ký tự mong muốn
    content = content.replace('&', 'and')
    content = content.replace('…', '...')
    content = content.replace(':.', ':')
    
    # Xóa các ký tự không phải là ASCII
    content = ''.join([char for char in content if char.isascii()])

    # Xóa ký tự '·' ở đầu chuỗi nếu có
    if content.startswith('·'):
        content = content[1:]

    # Xóa khoảng trắng thừa
    content = ' '.join(content.split())

    match = re.search(r"\.\s\d+\.$", content)
    if match:
        content = re.sub(r"\.\s\d+\.$", ".", content)

    return content

# Use tqdm to create a progress bar
data = df['input']
data_processed = []

for content in tqdm(data):
    cleaned_content = clean_text(content)
    if cleaned_content:
        data_processed.append(cleaned_content)

# # Nếu muốn lưu lại kết quả vào DataFrame mới
# df_cleaned = pd.DataFrame(data_processed, columns=['input'])

# # Hoặc lưu vào cột mới của DataFrame hiện tại
# df['cleaned_input'] = data_processed


100%|██████████| 20308/20308 [00:00<00:00, 43049.96it/s]


In [86]:
len(data_processed)

20308

In [87]:
df['input'] = data_processed

In [88]:
# data_processed

In [89]:
df

input  \
0      For example, countries with a lot of deserts c...   
1      As the number of people grows, the need of hab...   
2      Besides some technologically determinists that...   
3      Safety is one of the crucial problems that man...   
4      On one hand more and more virus and hack can a...   
...                                                  ...   
20303  Jane and monica, I receive the letter, first y...   
20304  against THE CINEMA, WE HAVE SEEN A COMEDY ENTI...   
20305  The romance is just a small parts where Batman...   
20306  Then, concerning a slight rising in 1992, it f...   
20307  Most of this were down to the speed following ...   

                                                  output  
0      For example, countries with a lot of deserts c...  
1      As the number of people grows, the need for a ...  
2      Besides some technological determinists that a...  
3      Safety is one of the crucial problems that man...  
4      On the one hand, more and more viruses and hac...  
...                                                  ...  
20303  Dear Jane and Monica, I have received the lett...  
20304  AT THE CINEMA, WE SAW A COMEDY ENTITLED " AUTU...  
20305  The romance is just a small part where Batman ...  
20306  Then, after a slight rise in 1992, they fell b...  
20307  Most of this is down to the speed at which our...  

[20308 rows x 2 columns]

In [90]:
dataset['train'] = Dataset.from_pandas(df.copy())

In [91]:
dataset

DatasetDict({
    train: Dataset({
        features: ['input', 'output'],
        num_rows: 20308
    })
    validation: Dataset({
        features: ['_id', 'task', 'src', 'tgt'],
        num_rows: 1712
    })
})

In [92]:
dataset['test'] = dataset['train'].select(range(17000,20308))
dataset['validation'] = dataset['train'].select(range(14000,17000))
dataset['train'] = dataset['train'].select(range(14_000))

In [93]:
dataset

DatasetDict({
    train: Dataset({
        features: ['input', 'output'],
        num_rows: 14000
    })
    validation: Dataset({
        features: ['input', 'output'],
        num_rows: 3000
    })
    test: Dataset({
        features: ['input', 'output'],
        num_rows: 3308
    })
})

In [94]:
dataset["train"][-9]

{'input': 'Seoul with Hiroaki Takeuchi, who is 26 years old, from 20th to 30th of next month.',
 'output': 'Seoul with Hiroaki Takeuchi, who is 26 years old, from the 20th to the 30th of next month.'}

# Evaluate

In [95]:
from transformers import pipeline

corrector = pipeline("text2text-generation", model="sehilnlf/model_v6")

In [96]:
dataset["test"][1]['input']

'However, my work is fun and worth for society, I think.'

In [97]:
inputs = []
outputs = []
for i in range(len(dataset["test"]['input'])):
    inputs.append(dataset["test"][i]['input'])
    outputs.append([dataset["test"][i]['output']])

In [100]:
from transformers import pipeline
from tqdm import tqdm

# Initialize the corrector pipeline with GPU support
corrector = pipeline("text2text-generation", model="sehilnlf/model_v6", device=0)  # Use device=0 for the first GPU

MAX_LENGTH = 512

# Define the text samples
texts = inputs  # No need to copy if we are just using the inputs directly

# Batch prediction
predictions = corrector(texts, max_length=MAX_LENGTH)

predicts = []
# Print predictions
for text, pred in tqdm(zip(texts, predictions), total=len(texts)):
    predicts.append(pred['generated_text'])

  0%|          | 0/10 [00:00<?, ?it/s]


In [101]:
import evaluate

In [ ]:
# input - output
google_bleu = evaluate.load("google_bleu")
glue = 0
count = 0
for j in tqdm(range(len(inputs))):
    try:
        glue += google_bleu.compute(predictions=[inputs[j]], references=[[outputs[j][0]]])['google_bleu']
        count += 1
    except:
        print(j)
print(glue/count)

In [102]:
# predict - output
google_bleu = evaluate.load("google_bleu")
glue = 0
count = 0
for j in tqdm(range(len(predicts))):
    try:
        glue += google_bleu.compute(predictions=[predicts[j]], references=[[outputs[j][0]]])['google_bleu']
        count += 1
    except:
        print(j)
print(glue/count)

100%|██████████| 1/1 [00:01<00:00,  1.02s/it]

0.5212765957446809


# Test with proposed dataset

In [1]:
import pandas as pd
df = pd.read_excel('/kaggle/input/gec-test-dataset/proposed_dataset.xlsx')
df

Tobe                          Tense  \
0               He are happy.    She go to school every day.   
1             She am at home.       He don't like ice cream.   
2    They is playing outside.      They was happy yesterday.   
3             I is a teacher.    I goes to work on weekends.   
4           You am a student.  She didn't went to the party.   
..                        ...                            ...   
116                       NaN                            NaN   
117                       NaN                            NaN   
118                       NaN                            NaN   
119                       NaN                            NaN   
120                       NaN                            NaN   

                         Article                              Preposition  \
0                  I saw an cat.                        I am good in math   
1          He is a best student.                 She is interested on art   
2     She is a intelligent girl.                He is afraid from spiders   
3    I have a important meeting.                   They arrived at Monday   
4             She is a engineer.   I am looking for my keys since morning   
..                           ...                                      ...   
116                          NaN                                      NaN   
117                          NaN                                      NaN   
118                          NaN                                      NaN   
119                          NaN                                      NaN   
120                          NaN                                      NaN   

                        ADJ_ADV  \
0           She sings beautiful   
1                 He runs quick   
2     They did good on the test   
3                She spoke soft   
4    He plays the piano perfect   
..                          ...   
116                         NaN   
117                         NaN   
118                         NaN   
119                         NaN   
120                         NaN   

                                      Relative_pronoun  \
0              The girl who hair is long is my sister.   
1    The car which engine was repaired yesterday ru...   
2    The house which roof was damaged in the storm ...   
3    The man who job is to clean the office is very...   
4    The computer that screen is cracked needs to b...   
..                                                 ...   
116     The dog which its owner is my aunt is trained.   
117  The park that we went to it last weekend is br...   
118  The person who I met him yesterday is my cowor...   
119  The car that I drove it to the mountains is ru...   
120  The movie that we watched it at the cinema was...   

                                         Passive_voice  
0    The computer screen is cracked needs to be rep...  
1    The dress was designed by the famous designer ...  
2    The book was written by the famous author very...  
3                 The girl hair is long my sister she.  
4             I don't like movie directed by him that.  
..                                                 ...  
116                                                NaN  
117                                                NaN  
118                                                NaN  
119                                                NaN  
120                                                NaN  

[121 rows x 7 columns]

In [3]:
df = df.dropna()
df

Tobe                            Tense  \
0              He are happy.      She go to school every day.   
1            She am at home.         He don't like ice cream.   
2   They is playing outside.        They was happy yesterday.   
3            I is a teacher.      I goes to work on weekends.   
4          You am a student.    She didn't went to the party.   
..                       ...                              ...   
84      The table are clean.     They were finished the task.   
85  The restaurant are open.             I do not goes there.   
86       The food are tasty.    He did not goes to the party.   
87     The weather are cold.  They is preparing for the exam.   
88        The sky are clear.   I doesn't know how to proceed.   

                                          Article  \
0                                   I saw an cat.   
1                           He is a best student.   
2                      She is a intelligent girl.   
3                     I have a important meeting.   
4                              She is a engineer.   
..                                            ...   
84  We had an wonderful time during our vacation.   
85        She has an beautiful voice for singing.   
86          I need an advice from a professional.   
87      We visited the Eiffel Tower in the Paris.   
88         She has a beautiful eyes that sparkle.   

                                       Preposition  \
0                                I am good in math   
1                         She is interested on art   
2                        He is afraid from spiders   
3                           They arrived at Monday   
4           I am looking for my keys since morning   
..                                             ...   
84           They walked along the beach on sunset   
85              I'm interested at learning Spanish   
86  He comes from a small town close the mountains   
87          We played soccer on the park yesterday   
88            He works hard to getting a promotion   

                                          ADJ_ADV  \
0                             She sings beautiful   
1                                   He runs quick   
2                       They did good on the test   
3                                  She spoke soft   
4                      He plays the piano perfect   
..                                            ...   
84                    She spoke softly than usual   
85              She walks slowlier in the morning   
86              She sings beautiful like an angel   
87          He runs more quicker than his brother   
88   I felt too terrible about the small accident   

                                     Relative_pronoun  \
0             The girl who hair is long is my sister.   
1   The car which engine was repaired yesterday ru...   
2   The house which roof was damaged in the storm ...   
3   The man who job is to clean the office is very...   
4   The computer that screen is cracked needs to b...   
..                                                ...   
84  The park that we visited it last summer is ama...   
85         The person who I talked to him is my boss.   
86        The car that I drove it on vacation is new.   
87  The movie that we watched it together was roma...   
88     The teacher who she teaches history is strict.   

                                        Passive_voice  
0   The computer screen is cracked needs to be rep...  
1   The dress was designed by the famous designer ...  
2   The book was written by the famous author very...  
3                The girl hair is long my sister she.  
4            I don't like movie directed by him that.  
..                                                ...  
84                   The window is break by the ball.  
85                        The book is readed by Mary.  
86    The house is build by the construction company.  
87                    The song is sing by the singer.  
88                 

In [7]:
89*7

623

In [14]:
from transformers import pipeline
from tqdm import tqdm

# Initialize the corrector pipeline with GPU support
corrector = pipeline("text2text-generation", model="sehilnlf/model_v6", device=0)  # Use device=0 for the first GPU

MAX_LENGTH = 512

# # Define the text samples
# texts = inputs  # No need to copy if we are just using the inputs directly

# # Batch prediction
# predictions = corrector(texts, max_length=MAX_LENGTH)

# predicts = []
# # Print predictions
# for text, pred in tqdm(zip(texts, predictions), total=len(texts)):
#     predicts.append(pred['generated_text'])

In [12]:
df.columns

Index(['Tobe', 'Tense', 'Article', 'Preposition', 'ADJ_ADV',
       'Relative_pronoun', 'Passive_voice'],
      dtype='object')

In [18]:
list(df[origin_name].values)

['He are happy.',
 'She am at home.',
 'They is playing outside.',
 'I is a teacher.',
 'You am a student.',
 'We is excited for the trip.',
 'It are raining today.',
 'This books are interesting.',
 'That car am expensive.',
 'These apples is red.',
 'Those houses am old.',
 'My friend are coming over.',
 'The children is playing in the park.',
 'The cake am delicious.',
 'The teacher are strict.',
 'The weather am hot today.',
 'Her dress are beautiful.',
 'The coffee are cold.',
 'The flowers is blooming.',
 'Our dog am very friendly.',
 'His car are new.',
 'The cat are sleeping.',
 'Their garden is beautiful.',
 'This pen are yours.',
 'His idea are interesting.',
 'Our house am cozy.',
 'Your advice are helpful.',
 'The soup are hot.',
 'The chair are comfortable.',
 'The team are winning.',
 'The book are easy to read.',
 'Her shoes is stylish.',
 'The painting are beautiful.',
 'His job are challenging.',
 'The restaurant are popular.',
 'Your car are parked outside.',
 'The ca

In [22]:
result = pd.DataFrame()
# origin_name = 'Tobe'

for origin_name in df.columns:
    texts = list(df[origin_name].values)

    # Batch prediction
    predictions = corrector(texts, max_length=MAX_LENGTH)

    predicts = []
    for text, pred in tqdm(zip(texts, predictions), total=len(texts)):
        predicts.append(pred['generated_text'])
        
    result[origin_name + '_corrected'] = predicts
# result.columns = [origin_name + '_corrected' for origin_name in df.columns]
result


100%|██████████| 89/89 [00:00<00:00, 507191.65it/s]


Tobe_corrected                   Tense_corrected  \
0                He is happy.     She goes to school every day.   
1             She is at home.        He doesn't like ice cream.   
2   They are playing outside.        They were happy yesterday.   
3             I am a teacher.     I go to work on the weekends.   
4          You are a student.       She didn't go to the party.   
..                        ...                               ...   
84        The table is clean.       They had finished the task.   
85    The restaurant is open.                I do not go there.   
86         The food is tasty.       He did not go to the party.   
87       The weather is cold.  They are preparing for the exam.   
88          The sky is clear.      I don't know how to proceed.   

                               Article_corrected  \
0                                   I saw a cat.   
1                          He is a best student.   
2                    She is an intelligent girl.   
3                   I have an important meeting.   
4                            She is an engineer.   
..                                           ...   
84  We had a wonderful time during our vacation.   
85        She has a beautiful voice for singing.   
86            I need advice from a professional.   
87         We visited the Eiffel Tower in Paris.   
88          She has beautiful eyes that sparkle.   

                                Preposition_corrected  \
0                                  I am good at math.   
1                           She is interested in art.   
2                            He is afraid of spiders.   
3      They arrived at the airport on Monday morning.   
4      I have been looking for my keys since morning.   
..                                                ...   
84             They walked along the beach at sunset.   
85                I'm interested in learning Spanish.   
86  He comes from a small town close to the mounta...   
87            We played soccer on the park yesterday.   
88                  He works hard to get a promotion.   

                                ADJ_ADV_corrected  \
0                      She sings beautiful songs.   
1                                  He runs quick.   
2                      They did good on the test.   
3                                 She spoke soft.   
4                     He plays the piano perfect.   
..                                            ...   
84                   She spoke softly than usual.   
85               She walks slower in the morning.   
86             She sings beautiful like an angel.   
87         He runs more quicker than his brother.   
88  I felt too terrible about the small accident.   

                           Relative_pronoun_corrected  \
0           The girl who's hair is long is my sister.   
1   The car which was repaired yesterday runs smoo...   
2   The house whose roof was damaged in the storm ...   
3   The man who's job is to clean the office is ve...   
4   The computer that is cracked needs to be repla...   
..                                                ...   
84   The park that we visited last summer is amazing.   
85             The person who I talked to is my boss.   
86           The car that I drove on vacation is new.   
87   The movie that we watched together was romantic.   
88     The teacher who she teaches history is strict.   

                              Passive_voice_corrected  
0   The computer screen is cracked and needs to be...  
1   The dress was designed by a famous designer an...  
2   The book was written by a famous author, a ver...  
3             The girl's hair is long, my sister she.  
4      I don't like movies directed by him that much.  
..                                                ...  
84                  The window is broken by the ball.  
85                          The book is read by Mary.  
86    The house is built by the construction company.  
87                

In [23]:
result

Tobe_corrected                   Tense_corrected  \
0                He is happy.     She goes to school every day.   
1             She is at home.        He doesn't like ice cream.   
2   They are playing outside.        They were happy yesterday.   
3             I am a teacher.     I go to work on the weekends.   
4          You are a student.       She didn't go to the party.   
..                        ...                               ...   
84        The table is clean.       They had finished the task.   
85    The restaurant is open.                I do not go there.   
86         The food is tasty.       He did not go to the party.   
87       The weather is cold.  They are preparing for the exam.   
88          The sky is clear.      I don't know how to proceed.   

                               Article_corrected  \
0                                   I saw a cat.   
1                          He is a best student.   
2                    She is an intelligent girl.   
3                   I have an important meeting.   
4                            She is an engineer.   
..                                           ...   
84  We had a wonderful time during our vacation.   
85        She has a beautiful voice for singing.   
86            I need advice from a professional.   
87         We visited the Eiffel Tower in Paris.   
88          She has beautiful eyes that sparkle.   

                                Preposition_corrected  \
0                                  I am good at math.   
1                           She is interested in art.   
2                            He is afraid of spiders.   
3      They arrived at the airport on Monday morning.   
4      I have been looking for my keys since morning.   
..                                                ...   
84             They walked along the beach at sunset.   
85                I'm interested in learning Spanish.   
86  He comes from a small town close to the mounta...   
87            We played soccer on the park yesterday.   
88                  He works hard to get a promotion.   

                                ADJ_ADV_corrected  \
0                      She sings beautiful songs.   
1                                  He runs quick.   
2                      They did good on the test.   
3                                 She spoke soft.   
4                     He plays the piano perfect.   
..                                            ...   
84                   She spoke softly than usual.   
85               She walks slower in the morning.   
86             She sings beautiful like an angel.   
87         He runs more quicker than his brother.   
88  I felt too terrible about the small accident.   

                           Relative_pronoun_corrected  \
0           The girl who's hair is long is my sister.   
1   The car which was repaired yesterday runs smoo...   
2   The house whose roof was damaged in the storm ...   
3   The man who's job is to clean the office is ve...   
4   The computer that is cracked needs to be repla...   
..                                                ...   
84   The park that we visited last summer is amazing.   
85             The person who I talked to is my boss.   
86           The car that I drove on vacation is new.   
87   The movie that we watched together was romantic.   
88     The teacher who she teaches history is strict.   

                              Passive_voice_corrected  
0   The computer screen is cracked and needs to be...  
1   The dress was designed by a famous designer an...  
2   The book was written by a famous author, a ver...  
3             The girl's hair is long, my sister she.  
4      I don't like movies directed by him that much.  
..                                                ...  
84                  The window is broken by the ball.  
85                          The book is read by Mary.  
86    The house is built by the construction company.  
87                

In [27]:
result

Tobe_corrected                   Tense_corrected  \
0                He is happy.     She goes to school every day.   
1             She is at home.        He doesn't like ice cream.   
2   They are playing outside.        They were happy yesterday.   
3             I am a teacher.     I go to work on the weekends.   
4          You are a student.       She didn't go to the party.   
..                        ...                               ...   
84        The table is clean.       They had finished the task.   
85    The restaurant is open.                I do not go there.   
86         The food is tasty.       He did not go to the party.   
87       The weather is cold.  They are preparing for the exam.   
88          The sky is clear.      I don't know how to proceed.   

                               Article_corrected  \
0                                   I saw a cat.   
1                          He is a best student.   
2                    She is an intelligent girl.   
3                   I have an important meeting.   
4                            She is an engineer.   
..                                           ...   
84  We had a wonderful time during our vacation.   
85        She has a beautiful voice for singing.   
86            I need advice from a professional.   
87         We visited the Eiffel Tower in Paris.   
88          She has beautiful eyes that sparkle.   

                                Preposition_corrected  \
0                                  I am good at math.   
1                           She is interested in art.   
2                            He is afraid of spiders.   
3      They arrived at the airport on Monday morning.   
4      I have been looking for my keys since morning.   
..                                                ...   
84             They walked along the beach at sunset.   
85                I'm interested in learning Spanish.   
86  He comes from a small town close to the mounta...   
87            We played soccer on the park yesterday.   
88                  He works hard to get a promotion.   

                                ADJ_ADV_corrected  \
0                      She sings beautiful songs.   
1                                  He runs quick.   
2                      They did good on the test.   
3                                 She spoke soft.   
4                     He plays the piano perfect.   
..                                            ...   
84                   She spoke softly than usual.   
85               She walks slower in the morning.   
86             She sings beautiful like an angel.   
87         He runs more quicker than his brother.   
88  I felt too terrible about the small accident.   

                           Relative_pronoun_corrected  \
0           The girl who's hair is long is my sister.   
1   The car which was repaired yesterday runs smoo...   
2   The house whose roof was damaged in the storm ...   
3   The man who's job is to clean the office is ve...   
4   The computer that is cracked needs to be repla...   
..                                                ...   
84   The park that we visited last summer is amazing.   
85             The person who I talked to is my boss.   
86           The car that I drove on vacation is new.   
87   The movie that we watched together was romantic.   
88     The teacher who she teaches history is strict.   

                              Passive_voice_corrected  
0   The computer screen is cracked and needs to be...  
1   The dress was designed by a famous designer an...  
2   The book was written by a famous author, a ver...  
3             The girl's hair is long, my sister she.  
4      I don't like movies directed by him that much.  
..                                                ...  
84                  The window is broken by the ball.  
85                          The book is read by Mary.  
86    The house is built by the construction company.  
87                

In [28]:
result.to_csv('/kaggle/working/corrected_result.csv')

In [29]:
88/89

0.9887640449438202

After using Grammarly to check the result of my model prediction. The result:
1. Tobe: 89/89 ~ 100%
2. Tense: 88/89 ~ 98.88%
3. Article: 79/89 ~ 88.76%
4. Preposition: 63/89 ~ 70.79%
5. ADJ_ADV: 32/89 ~ 35.96%
6. Relative_pronoun: 63/89 ~ 70.79%
7. Passive_voice: 69/89 ~ 77.53%
